server/api/student/student.controller.js

```diff
@@ -84,6 +87,123 @@ exports.outreachCounts = function(req, res) {
   });
 };
 
+function createNameAndSchoolPromise(results) {
+  return new Promise(function(resolve, reject) {
+    var promises = _.map(results, function(result) {
+      return new Promise(function(resolve, reject) {
+        function getTotals() {
+          var records = this.records;
+          var typeArr = _.map(records, 
+              function(rec) { return rec.type 
+            });
+          return _.countBy(typeArr, _.identity);
+        }
+        School.findById(result.school, 
+          function(err, school) {
+            if (err) reject(err);
+            result.school = school;
+          })
+          .then(function() {
+            Student.findById(result.student, 
+              function(err, student) {
+              if (err) reject(err);
+              result.student = student;
+            })
+          .then(function() {
+            result.totals = getTotals.call(result);
+            resolve(result);
+          });
+        });
+      });
+    });
+    Promise.all(promises).then(
+      function(val) { 
+        resolve(val);
+    });
+  });
+}
+
+/**
+ * Get summary of outreaches.
+ * restriction: 'teacher'
+ *
+ * Returns an aggregation for records based on the req user role:
+ * - teachers will get outreach summary for assignment school
+ * - manager+ will get outreach summary for all schools
+ */
+exports.outreachSummary = function(req, res) {
+  var pipeline = [{
+    $group: {
+      _id: {student: '$student', school: '$school'},
+      records: { $addToSet: '$$ROOT' }
+    } 
+  }, {
+    $project: {
+      _id: 0,
+      school: '$_id.school',
+      student: '$_id.student',
+      records: {
+        type: 1,
+        tier: 1,
+        absences: 1,
+        record: 1,
+        schoolYear: 1,
+        triggerDate: 1,
+        actionDate: 1
+      }
+    }
+  }];
+  if(req.user.role === 'teacher') {
+    pipeline.push({ 
```

Comment:
For aggregations the order that these pipeline items will make a big difference. If you push the match to the end you lose out on reducing the number of operations that the aggregation has to do.